spec/docker-compose-dev.yml

```diff
@@ -0,0 +1,26 @@
+version: "3"
+services:
+  linuxdev:
+    hostname: linuxdev
+    domainname: bolt.test
+    # must use Samba AD for DNS
+    dns:
+      - 172.22.0.100
+      - 8.8.8.8
+    build:
+      context: .
+      dockerfile: Dockerfile.linuxdev
+      args:
+        BOLT_PASSWORD: bolt
+    ports:
+      - "422:22"
+    environment:
+      KRB5_REALM: ${KRB5_REALM:-BOLT.TEST}
+      KRB5_KDC: ${KRB5_KDC:-samba-ad.bolt.test}
+      SMB_DOMAIN: ${SMB_DOMAIN:-BOLT}
+      SMB_ADMIN: Administrator
+      SMB_ADMIN_PASSWORD: ${SMB_ADMIN_PASSWORD:-B0ltrules!}
+    depends_on:
+      - samba-ad
```

Comment:
I think since this isn't defined in this file it doesn't know about it. Is there a reason not to have this in the main `docker-compose`?

Reply:
The deal there is that I didn’t want to build that container unless it was explicitly asked for since it’s a dev container.. hence the separate file.

Dockers `depends_on` doesn’t actually guarantee anything useful in terms of orchestrating containers - it only agrees to start `samba-ad` first. Since our instructions already start the other compose file, before starting this one, then that requirement is already met.


IOW, fine to remove it I think.

Reply:
Actually on second thought... can you verify you actually ran the commands as documented?

> docker-compose -f spec/docker-compose.yml -f spec/docker-compose-dev.yml build --build-arg BUILD_OMI=true samba-ad omiserver linuxdev
> docker-compose -f spec/docker-compose.yml -f spec/docker-compose-dev.yml up samba-ad omiserver linuxdev

Multiple YAML files should get merged prior to running anything.

If you tried to run just `docker-compose-dev.yml` by itself I could see why things might fail.